front50-proto/src/main/proto/applications.proto

```diff
@@ -0,0 +1,241 @@
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+package spinnaker.front50.v3;
+
+option java_package = "com.netflix.spinnaker.front50.proto";
+option java_multiple_files = true;
+
+// ApplicationService is the service for interacting with the Spinnaker application domain.
+service ApplicationService {
+  // ListApplications provides a way of searching through Applications..
+  rpc ListApplications (ListApplicationsRequest) returns (ListApplicationsResponse);
+
+  // GetApplication returns a single Application matching the criteria provided.
+  rpc GetApplication (GetApplicationRequest) returns (Application);
+
+  // CreateApplication will create a new Application. If an Application already exists by its name, this operation
+  // will fail.
```

Comment:
We should probably be specific about the type of [errors](https://www.grpc.io/docs/guides/error/) we should raise. Something like the precondition failure seems appropriate here? https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto#L93-L115

Reply:
Yes, agree - from what I understand of gRPC (which is admittedly not much yet), these effort formats are not actually defined as part of the response, but instead handled through exceptions. Isn't that correct? I'm operating off of this page: https://cloud.google.com/apis/design/errors

Reply:
I think you are correct (also not a gRPC expert). This isn't explicit in the response object. When documenting the API, we should probably be specific about particular anticipated errors.

Reply:
💯 on the documentation side of things. Since it's not part of the payload, I'm kinda biased towards documenting in a single location similar to the links we provided each other; rather than mentioning in each proto that X proto can return Y, Z, N... errors. WDYT?

Reply:
For this particular comment, I'd prefer moving:

> this operation will fail.

to

> this operation will raise a PreconditionFailure.

Reply:
I don't think an exhaustive list is necessary, but precision in particularly common exceptions we anticipate seems worthwhile.

Reply:
Thanks, makes sense, appreciate the clarification.